queue/queue_private.js

```diff
@@ -1,63 +1,39 @@
-
-$(document).ready(refresh);
-
-$('.date-wrap').on('click', '.date-back', hideDetails);
-$('.date-wrap').on('click', '.date-forward', hideDetails);
-$('.details-controls').on('click', '.details-close', hideDetails);
-
-// Cancels the selected appointment
-$('.details-controls').on('click', '.details-cancel', function() {
-	if (confirm('Are you sure you would like to cancel this appointment? Click OK to cancel this appointment.')) {
-		$.get({
-			data: `id=${openedAppointmentId}&action=cancel`,
-			url: '/server/queue_priv.php',
-			success: function(r) {
-				hideDetails();
-				populateQueue();
-			}
-		});
-	}
+queueApp.controller("QueuePrivateController", function($scope, $controller, QueueService) {
+	angular.extend(this, $controller('QueueController', {$scope: $scope}));
+
+	$scope.selectClient = function(client) {
+		$scope.client = client;
+	};
 });
 
-// Retrieves and displays the data associated with the selected appointment
-$('.queue-table').on('click', '.queue-record', function() {
-	openedAppointmentId = $(this).data('appointmentId');
-
-	// Adds a soft highlight to the last selected appointment
-	$('.queue-record').removeClass('theme-light');
-	$(this).addClass('theme-light');
-
-	$.get({
-		data: `id=${openedAppointmentId}&action=display`,
-		url: '/server/queue_priv.php',
-		dataType: 'json',
-		cache: false,
-		success: function(r) {
-			let t = new Date(r[0].scheduledTime), h = t.getHours() % 12, m = t.getMinutes();
-			if (h === 0) h = 12;
-			if (m < 10) m = `0${m}`;
-
-			$('.details-name').html(`${r[0].firstName} ${r[0].lastName}`);
-			$('.details-email').html(r[0].emailAddress ? r[0].emailAddress : 'None');
-			$('.details-phone').html(r[0].phoneNumber ? r[0].phoneNumber : 'None');
-			$('.details-site-name').html(r[0].title);
-			$('.details-time').html(`${h}:${m}`);
-			$('.details').css('display', 'flex');
+queueApp.filter('searchFor', function(){
+
+	// All filters must return a function. The first parameter
+	// is the data that is to be filtered, and the second is an
+	// argument that may be passed with a colon (searchFor:searchString)
+
+	return function(arr, searchString){
+
+		if(!searchString){
+			return arr;
 		}
-	});
-});
 
-function refresh() {
-	displayQueueDate();
-	populateQueue();
+		var result = [];
 
-	clearTimeout(refreshing);
-	refreshing = setTimeout(refresh, REFRESH_SEC * 1000);
-}
+		searchString = searchString.toLowerCase();
 
-function hideDetails() {
-	$('.queue-record').removeClass('theme-light');
-	$('.details').css('display', 'none');
-	openedAppointmentId = 0;
-}
\ No newline at end of file
+		// Using the forEach helper method to loop through the array
+		angular.forEach(arr, function(item){
+
+			if(item.name.toLowerCase().indexOf(searchString) !== -1 ||
+				 item.appointmentId.toString().indexOf(searchString) !== -1 ||
+				 ("#" + item.appointmentId.toString()).indexOf(searchString) !== -1){
+				result.push(item);
+			}
```

Comment:
Wow this search feature is nice and surprisingly simple to implement. Thank you AngularJS. And also, good call on including the AppointmentId on the page and in this search feature.